### PR TITLE
Setup project

### PR DESCRIPTION
**Pull request by perryliao**

### CODE REVIEW

File: app/src/components/EnhancedComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import * as React from "react";
+import {ReactNode} from "react";
+
+abstract class EnhancedComponent<P extends IEnhancedComponentProps = IEnhancedComponentProps, S extends IEnhancedComponentState = IEnhancedComponentState> extends React.PureComponent<P, S> {
```
Comment (aditya10) [inline app/src/components/EnhancedComponent.tsx:4]: Typescript makes this look more complicated than necessary, but I guess that's an ok tradeoff

File: app/src/containers/App.css [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+@keyframes App-logo-spin {
+  from {
+    transform: rotate(0deg);
+  }
+  to {
+    transform: rotate(360deg);
+  }
+}
```
Comment (aditya10) [inline app/src/containers/App.css:38]: Is our folder structure going to be 
```
src/components/<componentName>/
* <componentName>.tsx
* <componentName>.test.tsx
* <componentName>.css
```
Each component would have its test and css file in the same folder with it? (I like this structure)

File: app/src/components/EnhancedComponent.tsx (hunk 1 above)
Comment (perryliao) [reply app/src/components/EnhancedComponent.tsx:4]: it's just these 2 files that'll have it look like this, others will be shorter

File: app/src/containers/App.css (hunk 2 above)
Comment (perryliao) [reply app/src/containers/App.css:38]: that works, or if you have simple css you could just make a style object in its class, which is actually what I'm used to using haha

**Comment by perryliao** [comment]
Merging!

**Pull request merged**